tssc/step_implementers/unit_test/junit.py

```diff
@@ -107,8 +118,74 @@ def _run_step(self, runtime_step_config):
         dict
             Results of running this step.
         """
-        results = {}
+        pom_file = runtime_step_config['pom-file']
+
+        if not os.path.exists(pom_file):
+            raise ValueError('Given pom file does not exist: ' + pom_file)
+
+        maven_surefire_plugin = self.find_reference_in_pom(pom_file, 'maven-surefire-plugin')
+        if maven_surefire_plugin is None:
+            raise ValueError('Unit test dependency "maven-surefire-plugin" missing from POM.')
+
+        reports_dir = self.find_reference_in_pom(pom_file, 'reportsDirectory')
+        if reports_dir is not None:
+            default_test_results_dir = reports_dir
+        else:
+            default_test_results_dir = os.path.join(
+                os.path.dirname(os.path.abspath(pom_file)),
+                'target/surefire-reports')
+
+        test_results_output_path = self._StepImplementer__results_dir_path + '/unit-test/junit' # pylint: disable=no-member
+
+        try:
+            sh.mvn(  # pylint: disable=no-member
+                'clean',
+                'test',
+                '-f', pom_file,
+                _out=sys.stdout
+            )
+        except sh.ErrorReturnCode as error:
+            raise RuntimeError("Error invoking mvn: {error}".format(error=error))
+
+        if not os.path.isdir(default_test_results_dir) or \
+            len(os.listdir(default_test_results_dir)) == 0:
+            test_results_output_path = "NO UNIT TEST RESULTS"
```

Comment:
@HunterGerlach shoulnd't this be an error?

Reply:
No, not necessarily. It could be a byproduct of having no test classes. Just because you tell it to run `mvn test` and the appropriate plugins are there, doesn't mean there are tests to run against. Since the unit-test step is doing it's job correctly, I don't this this is a failure, but should notify the user that there are no results to report.

Reply:
hmmmm....i know its a bit of scope crepe but could we add flag like `fail-on-no-tests` or something and if true it throws an error in this case and if false just logs like you do here?

Reply:
i can just sense some compliance person being like, i want to mandate that unit tests are actaully run. and at some point we probably have to deal with test coverage......but i guess thats another jira...

Reply:
Added fail-on-no-tests. Marking this as resolved.